Lesson17/main.js

```diff
@@ -0,0 +1,124 @@
+const slideList = document.getElementById("js-slid-list");
+const slideWrapper = document.getElementById("js-slide-wrapper");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+}
+
+const addLoading = (parent) => {
+  const loading = createElementWithClassName("img", "loading");
+  loading.id = "js-loading";
+  loading.src = "./loading-circle.gif";
+  parent.appendChild(loading);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const addErrorMessage = (error, parent) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  parent.appendChild(errorMessage);
+}
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+const getSlideImageData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/gzcn")), 3000);
+  })
+}
+
+const tryGetSlideImageData = async () => {
+  addLoading(slideWrapper);
+  try {
+    return await getSlideImageData();
+  } catch (e) {
+    addErrorMessage(e, slideWrapper);
+  } finally {
+    removeLoading();
+  }
+}
+
+const init = async () => {
+  const imageData = await tryGetSlideImageData();
+  if (imageData) {
+    for (let i = 0; i < imageData.length; i++) {
+      const createdSlideItem = createSlideItem(imageData[i]);
+      //初期設定として 最初の要素にis-displayingクラスを付与
+      i === 0 && createdSlideItem.classList.add("is-displaying");
+      addElement(createdSlideItem, slideList);
+    }
```

Comment:
IMO
I felt that this part could be separated into functions.

Reply:
Thanks for the review.
I'm always wondering where to make the functions...

I divided it into two functions: one to initialize the slide item, and the other to initialize the arrow button.
e7e6bd5